tools/nni_cmd/launcher.py

```diff
@@ -75,24 +75,40 @@ def start_rest_server(port, platform, mode, config_file_name, experiment_id=None
 
     print_normal('Starting restful server...')
     # Find nni lib from the following locations in order
-    sys_wide_python = True
-    python_sitepackage = site.getsitepackages()[0]
-    # If system-wide python is used, we will give priority to using user-sitepackage given that nni exists there
-    if python_sitepackage.startswith('/usr') or python_sitepackage.startswith('/Library'):
-        local_python_dir = str(Path(site.getusersitepackages()).parents[2])
-        entry_file = os.path.join(local_python_dir, 'nni', 'main.js')
-        entry_dir = os.path.join(local_python_dir, 'nni')
-    else:
-        # If this python is not system-wide python, we will use its site-package directly
-        sys_wide_python = False
-
-    if not sys_wide_python or not os.path.isfile(entry_file):
-        python_dir = str(Path(python_sitepackage).parents[2])
+    def get_installation_path(which_first):
+        # 0 means `local sitepackages`--"usersitepackages()" is checked first while 1 is the opposite 
+        def _generate_installation_path(sitepackages_path):
+            python_dir = str(Path(sitepackages_path).parents[2])
+            entry_file = os.path.join(python_dir, 'nni', 'main.js')
+            if os.path.isfile(entry_file):
+                return python_dir
+            return None
+
+        sitepackages = [site.getusersitepackages(), site.getsitepackages()[0]]
+        python_dir = _generate_installation_path(sitepackages[which_first])
+        if python_dir is not None:
+            return python_dir
+        return _generate_installation_path(sitepackages[1-which_first])
```

Comment:
Not easy to understand,  suggest to refactor this piece to make it more straightforward.     For example maybe remove the which_first param  and directly pass sitepackages...

Reply:
Good suggestion. Has modified. How about now?

Reply:
1. It is much better now.  Since the start_rest_server is already too long.  It would be better if you extract the logic to acquire entry_file into a seperate function.
2. It seems the entry_dir is not used.

Reply:
- Good suggestion. Has move that piece of code to a new function
- `entry_dir` is used when invoking Popen in the next several lines so I kept it